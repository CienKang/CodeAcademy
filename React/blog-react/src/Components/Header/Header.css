header {
    height: 12vh;
    background-color: black;
    color: white;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;

}

header .name-div {
    display: flex;
    flex-direction: column;
    gap: 2px;
    margin-left: 15vw;
}

.name-div h1 {
    font-size: 1.6rem;
    font-weight: lighter;
}

.name-div h2 {
    font-size: 1.9rem;
    font-weight: lighter;
}

.link-bar-div {
    align-self: flex-end;
    display: flex;
    gap: 10vw;
    margin-right: 15vw;
    margin-bottom: 5px;
}

.link-bar-div span {
    font-size: 1.2rem;
    font-weight: lighter;
}
