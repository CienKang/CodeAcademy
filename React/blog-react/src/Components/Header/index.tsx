import React from 'react';

import './Header.css';
const Header = (): JSX.Element => {
    return (
        <header>
            <div className="name-div outer-margin-left-15">
                <h2>The Artifact</h2>
                <h1><i>Culture & Art Blog</i></h1>
            </div>

            <div className="link-bar-div">
                <span>Blog</span>
                <span>About</span>
                <span>Contact</span>
            </div>
        </header>
    );
};

export default Header;