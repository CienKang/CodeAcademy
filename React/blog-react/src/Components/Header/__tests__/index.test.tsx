import React from 'react';
import {render} from '@testing-library/react';
import Header from '..';


describe('Header', () => {

    it('should render the header correctly when rendered for first time', () => {
        const {getByText} = render(<Header />);
        expect(getByText('The Artifact')).toBeTruthy();
        expect(getByText('Culture & Art Blog')).toBeTruthy();
        expect(getByText('Blog')).toBeTruthy();
        expect(getByText('About us')).toBeTruthy();
        expect(getByText('Contact')).toBeTruthy();
    });

    it('should match the snapshot ', () => {
        const {asFragment} = render(<Header />);
        expect(asFragment()).toMatchSnapshot();
    });
});