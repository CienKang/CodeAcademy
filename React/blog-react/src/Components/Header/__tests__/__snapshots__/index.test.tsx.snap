// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot  1`] = `
<DocumentFragment>
  <div>
    <div
      class="flex flex-col bg-black justify-center"
    >
      <h1
        class="text-white text-lg ml-32"
      >
        The Artifact
      </h1>
      <p
        class="italic text-white text-sm ml-32"
      >
        Culture & Art Blog
      </p>
    </div>
    <div
      class="bg-black"
    >
      <div
        class="flex flex-row items-center justify-end mr-20 space-x-10"
      >
        <a
          class="italic text-sm text-white"
          href=""
        >
          Blog
        </a>
        <a
          class="italic text-sm text-white"
          href=""
        >
          About us
        </a>
        <a
          class="italic text-sm text-white"
          href=""
        >
          Contact
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
